components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/CertificateValidationUtil.java

```diff
@@ -1494,4 +1411,454 @@ private static Validator resourceToValidatorObject(
+                    .updateCertificateContent(certificateId, encodeCertificate(certificate), tenantDomain);
+            log.debug("Successfully updated certificate with the id: " + certificateId + " in certificate manager.");
+            return decodeCertificate(deletedCertificate.getCertificateContent());
+        } catch (CertificateMgtException e) {
+            log.debug("Error when getting certificate from certificate manager.", e);
+            if (CertificateMgtErrors.ERROR_CERTIFICATE_DOES_NOT_EXIST.getCode().equals(e.getErrorCode())) {
+                throw new CertificateValidationException
+                        ("Certificate with the id: " + certificateId + " does not exist.", e);
+            } else {
+                throw new CertificateValidationException("Error when updating certificate in certificate manager.", e);
+            }
+        } catch (CertificateException e) {
+            log.debug("Error when decoding certificate.", e);
+            throw new CertificateValidationException("Error when encoding certificate.", e);
+        }
+    }
+
+    /**
+     * Delete a certificate in the configuration store by certificate id.
+     *
+     * @param certificateId Certificate Id
+     * @param tenantDomain Tenant Domain
+     * @return CertObject
+     * @throws CertificateValidationException Error when deleting certificate
+     */
+    public static CertObject deleteCertificateInConfigurationStoreByCertificateId(String certificateId,
+                                                                                  String tenantDomain)
+            throws CertificateValidationException {
+
+        try {
+            org.wso2.carbon.identity.configuration.mgt.core.model.Resource resource =
+                    CertValidationDataHolder.getInstance().getConfigurationManager()
+                    .getResourceByTenantId(IdentityTenantUtil.getTenantId(tenantDomain),
+                            X509_CA_CERT_RESOURCE_TYPE, CERTS);
+            InputStream inputStream = getResourceFileInputStream(resource);
+            if (inputStream == null) {
+                return null;
+            }
+
+            IssuerDNMap issuerDNMap = ModelSerializer.deserializeIssuerDNMap(convertInputStreamToString(inputStream));
+
+            for (Map.Entry<String, List<CertObject>> entry : issuerDNMap.getIssuerCertMap().entrySet()) {
+                List<CertObject> certList = entry.getValue();
+                if (certList.removeIf(cert -> cert.getCertId().equals(certificateId))) {
+                    if (certList.isEmpty()) {
+                        issuerDNMap.getIssuerCertMap().remove(entry.getKey());
+                    }
+                    return saveUpdatedResource(resource, issuerDNMap);
+                }
+            }
+            return null;
+        } catch (IOException | ConfigurationManagementException e) {
+            throw new CertificateValidationException("Error while processing the resource", e);
```

Comment:
Why are we not introducing error codes via defining error enums?

Reply:
Address via https://github.com/wso2/product-is/issues/22818